### PR TITLE
add demo tests

### PR DESCRIPTION
**Pull request by tgetachew18**
eat

### CODE REVIEW

File: test/PoolFactory.js [hunk 1]
```diff
@@ -8,13 +8,37 @@ describe("PoolFactory contract", function () {
 
   beforeEach(async function () {
     poolFactoryContract = await ethers.getContractFactory("PoolFactory");
-    [owner] = await ethers.getSigners();
+    [owner, addy1, addy2] = await ethers.getSigners();
```
Comment (ksgebrekirstos98) [inline test/PoolFactory.js:11]:
```suggestion
    [owner, address1, address2] = await ethers.getSigners();
```

File: test/PoolFactory.js [hunk 2]
```diff
@@ -8,13 +8,37 @@ describe("PoolFactory contract", function () {
...
-  // TODO: Add tests
+  
+  describe("Pool dynamics", function () {
+    // `it` is another Mocha function. This is the one you use to define your
+    // tests. It receives the test name, and a callback function.
+
+    // If the callback function is async, Mocha will `await` it.
+    it("Should create a pool", async function () {
+      //create a pool
+      poolRef = await poolFactoryToken.createPool("tadPool", [addy1.address]);
```
Comment (ksgebrekirstos98) [inline test/PoolFactory.js:25]: Love the idea but let's use real names :) 
```suggestion
      poolRef = await poolFactoryToken.createPool("TestPool1", [addy1.address]);
```

File: test/PoolFactory.js [hunk 3]
```diff
@@ -8,13 +8,37 @@ describe("PoolFactory contract", function () {
...
+      pools = await poolFactoryToken.listPools();
+      expect( pools.length).to.equal(1);
```
Comment (ksgebrekirstos98) [inline test/PoolFactory.js:27]: Can you add a case where you assert that the name of the pool matches the expected value?

File: test/PoolFactory.js [hunk 4]
```diff
@@ -8,13 +8,37 @@ describe("PoolFactory contract", function () {
...
+    });
+
+    it("Pool should have admin var defined", async function () {
```
Comment (ksgebrekirstos98) [inline test/PoolFactory.js:30]: Not sure I understand what this test is doing. Why is the admins length equal to 0?

Comment (tgetachew18) [reply test/PoolFactory.js:30]: nvm your suggestion of just usign the name makes more sense

File: test/PoolFactory.js [hunk 5]
```diff
@@ -8,13 +8,42 @@ describe("PoolFactory contract", function () {
...
+      // Acquire pool contract
+      poolName = "TestPool2";
+      admins = [admin1.address, admin2.address];
+      poolRef = await poolFactoryToken.createPool(poolName, admins);
+
+      poolConFac = await ethers.getContractFactory("Pool");
+      poolAddress = (await poolFactoryToken.listPools())[0];
+      poolCon = await poolConFac.attach(poolAddress);
+
+      gotName = await poolCon.name();
```
Comment (ksgebrekirstos98) [inline test/PoolFactory.js:40]: This is what you were trying to do right?
```suggestion
      expect(await poolCon.name).to.equal(poolName);
```

File: test/PoolFactory.js [hunk 6]
```diff
@@ -8,13 +8,42 @@ describe("PoolFactory contract", function () {
...
+      gotAdmins = await poolCon.admins;
```
Comment (ksgebrekirstos98) [inline test/PoolFactory.js:41]: This is not needed
```suggestion
```

**Review by ksgebrekirstos98 — APPROVED** [review]

**Pull request merged**